### DISCUSSION TITLE
How to use option '-sort', '-sortnode' and '-sortelt'

### DISCUSSION THREAD

**Discussion by damulaoshi** [neperfepx/neper#126] [Q&A] [answered]
Dear Romain and all Neper users:
First of all, I want to appreciate you, Dr. Romain Quey, for your contributions to this very useful software, it really helps me a lot during my research!

My question is how can I use the option '-sort' to facilitate data post-processing? From 'Neper Reference Manual', it just says that the possible value could be any. Is there any example that you can show me? 
(
If my code is like below in Module-T:
neper -T -n 100 -dim 3 -domain "cube(1,1,1)" -id 2 -statcell vol,diameq -ori random -sort (here I don't know how to define the value) -o delta1
)

Thank you and I am looking forward to your answer.
Best regards,
Damu

**Comment by rquey** [comment, marked as answer]
It can be `-sort z` to get grains sorted in ascending z coordinate, `-sort vol` for ascending volumes, `-sort x+2*z` for ascending x+2*z, etc.

**Reply by damulaoshi** [reply]
Thank you so much for answering so quickly! I will have a try!

**Reply by damulaoshi** [reply]
Dear Romain:
I have tried yesterday and it works! But I think I still have problems even though I use `-sort`, `-sortnode` or `-sortelt`. 

My case is like this:
I have a domain of `cube(1,1,1)` and I need to partition it into 8 same subdomains. I first generate the `cube(1,1,1)` then use `-transform "crop(cube(...))` to generate 8 subdomains. Then I do mesh for the whole domain and 8 subdomains separately. During the crop process in Module-T, some of the grains will be cut at the boundary surface of subdomains.

My question is that, can we know which grain has been cut during the crop process according to some code in Neper directly(in Module-T or in Module-M)? Right now I just can open the '.inp' file generated by Neper, and comparing them(the whole domain and subdomains) to see which grain has been cut to 2 or more parts during the 'crop' process. 

My code is like below:

```
$ neper -T -n 216 -dim 3 -domain "cube(1,1,1)" -id 2  -statcell vol,diameq -ori random -transform "crop(cube(0,0.5,0,0.5,0,0.5))" -sort id -o delta

$ neper -M delta.tess -order 1 -elttype hex -cl 0.05 -format msh,inp -statmesh nodenb -sortnode coo -sortelt elset3d -o delta6
```

I really need to know at the boundary, which grain has been cut when I use `-transform crop`, and I need to relate these cut grains of 8 subdomains (`-transform "crop(cube(0,0.5,0,0.5,0,0.5))"`, etc..) to the grains of the whole domain (`cube(1,1,1)`), because for these grains (which will be cut into 2 or more grains in subdomains, but in fact, in the whole domain some of them belongs to one grain), I need to define them the same properties when I do the post-processing.

Thank you so much and I am looking forward to your answer.
Best regards,
Damu

**Reply by rquey** [reply]
The grains retain their ids (and their other properties such as their orientations, etc.) when you do the `-transform crop`, so that you can very easily relate the grains of the whole domain to the grains of the subdomains.

As to how to determine which grain has been cut, an easy way is to use the volume...

**Reply by damulaoshi** [reply]
Ah! Ok now I know it! I'll have a try. Thank you so much!
